modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js

```diff
@@ -0,0 +1,201 @@
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayManagementToolbar from '@clayui/management-toolbar';
+import ClayModal from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const TranslationModal = ({
+	ariaLabels = {
+		default: Liferay.Language.get('default'),
+		manageTranslations: Liferay.Language.get('manage-translations'),
+		translated: Liferay.Language.get('translated'),
+		untranslated: Liferay.Language.get('untranslated'),
+	},
+	defaultLocale,
+	locales,
+	observer,
+	translations,
+}) => {
+	const [active, setActive] = React.useState(false);
+	const [icon, setIcon] = React.useState('search');
+	const inputRef = React.createRef();
+	const [visibleLocales, setVisibleLocales] = React.useState(locales);
+
+	return (
+		<ClayModal observer={observer}>
+			<ClayModal.Header>
+				{Liferay.Language.get('manage-translations')}
+			</ClayModal.Header>
+
+			<ClayModal.Body scrollable>
+				<ClayManagementToolbar aria-label="Management toolbar">
+					<ClayManagementToolbar.Search showMobile={true}>
+						<ClayInput.Group>
+							<ClayInput.GroupItem>
+								<ClayInput
+									aria-label={Liferay.Language.get('search')}
+									className="form-control input-group-inset input-group-inset-after"
+									onInput={() => {
+										const value = inputRef.current?.value;
+
+										const icon = value ? 'times' : 'search';
+										setIcon(icon);
+
+										if (!value) {
+											setVisibleLocales(locales);
+
+											return;
+										}
+
+										const result = Object.values(
+											visibleLocales
+										).filter((locale) => {
+											return locale.label.match(
+												new RegExp(value, 'i')
+											);
+										});
+
+										setVisibleLocales(result);
+									}}
+									placeholder={Liferay.Language.get('search')}
+									ref={inputRef}
+									type="text"
+								/>
+								<ClayInput.GroupInsetItem after tag="span">
+									<ClayButtonWithIcon
+										aria-label={Liferay.Language.get(
+											'search'
+										)}
+										displayType="unstyled"
+										onClick={() => {
+											if (
+												icon === 'times' &&
+												inputRef.current?.value
+											) {
+												inputRef.current.value = '';
+												setIcon('search');
+												setVisibleLocales(locales);
+											}
+										}}
+										symbol={icon}
+									/>
+								</ClayInput.GroupInsetItem>
+							</ClayInput.GroupItem>
+						</ClayInput.Group>
+					</ClayManagementToolbar.Search>
+
+					<ClayManagementToolbar.ItemList>
+						<ClayDropDown
+							active={active}
+							onActiveChange={setActive}
+							trigger={<ClayButtonWithIcon symbol="plus" />}
+						></ClayDropDown>
```

Comment:
Nothing too critical here, but it's more common to use more self-closing tag in JSX.

Reply:
True, I'll do that.

Reply:
Updated in [`9349512`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/93495121c4028424aee39bc9746b60aad2597387)